services/filemanager/s3manager.go

```diff
@@ -102,6 +82,35 @@ func (manager *S3Manager) Download(output *os.File, key string) error {
 	return nil
 }
 
+/*
+GetObjectNameFromLocation gets the object name/key name from the object location url
+	https://bucket-name.s3.amazonaws.com/key - >> key
+*/
+
+func (manager *S3Manager) GetObjectNameFromLocation(location string) (string, error) {
+	parsedUrl, err := url.Parse(location)
+	if err != nil {
+		return "", err
+	}
+	trimedUrl := strings.TrimLeft(parsedUrl.Path, "/")
+	if manager.Config.S3ForcePathStyle != nil && *manager.Config.S3ForcePathStyle {
```

Comment:
added check for `S3ForcePathStyle` flag. Since, if  `S3ForcePathStyle` is set to `true` then location url will of type: `http://localhost:55190/filemanager-test-1/some-prefix/original.json.gz`. And, if not set, then it will of type: `http://filemanager-test-1.localhost:55191/some-prefix/original.json.gz`. I.e. in 1st case bucketName if part of path, whereas in 2nd case, it is a part of host.

Reply:
Not sure if this happens only when `S3ForcePathStyle` is set. The `Force` makes me think that is if this is not provided we can not guarantee that will not be `S3PathStyle` ...

Reply:
Checkout the following: https://docs.aws.amazon.com/AmazonS3/latest/userguide/VirtualHosting.html

Reply:
I believe that the best way to determine if is path style is to check the domain name for the bucket.

If `S3ForcePathStyle` is set OR domain doesn't contain bucket name -> pathStyle

Reply:
changes done for this.